lazy_dataset/parallel_utils.py

```diff
@@ -123,14 +124,43 @@ def lazy_parallel_map(
     if backend == "mp":
         # http://stackoverflow.com/a/21345423
         from pathos.multiprocessing import ProcessPool as PathosPool
+        import pathos.helpers.pp_helper
         PoolExecutor = PathosPool
 
         def submit(ex, func, *args, **kwargs):
             return ex.apipe(func, *args, **kwargs)
 
+        def result(job: pathos.helpers.pp_helper.ApplyResult):
+            return job.get()
+
+        def terminate(ex: pathos.multiprocessing.ProcessPool, q):
+            ex.terminate()
+            # Cancel doesn't work for pathos. Don't know why.
+            # try:
+            #     while True:
+            #         q.get(block=False).cancel()
+            # except queue.Empty:
+            #     pass
+
+    elif backend == 'multiprocessing':
+        from multiprocessing import Pool as PoolExecutor
+
+        def submit(ex, func, *args, **kwargs):
+            return ex.apply_async(func, args, kwargs)
+
         def result(job):
             return job.get()
 
+        def terminate(ex, q):
+            # It looks like multiprocessing works fine with GeneratorExit
```

Comment:
From the docs:
> Stops the worker processes immediately without completing outstanding work

Reply:
Added a comment, why multiprocessing just works as it should.